lib/segment/src/index/sparse_index/sparse_vector_index.rs

```diff
@@ -107,14 +107,25 @@ impl<TInvertedIndex: InvertedIndex> SparseVectorIndex<TInvertedIndex> {
         let mut index_point_count: usize = 0;
         for id in borrowed_id_tracker.iter_ids_excluding(deleted_bitslice) {
             check_process_stopped(stopped)?;
-            let vector = borrowed_vector_storage.get_vector(id);
-            let vector: &SparseVector = vector.as_vec_ref().try_into()?;
-            // do not index empty vectors
-            if vector.is_empty() {
-                continue;
+            // It is possible that the vector is not present in the storage in case of crash.
+            // Because:
+            // - the `id_tracker` is flushed before the `vector_storage`
+            // - the sparse index is built *before* recovering the WAL when loading a segment
+            match borrowed_vector_storage.get_vector_opt(id) {
+                None => {
+                    // the vector was lost in a crash but will be recovered by the WAL
+                    log::debug!("Sparse vector with id {} is not found", id)
+                }
+                Some(vector) => {
+                    let vector: &SparseVector = vector.as_vec_ref().try_into()?;
+                    // do not index empty vectors
+                    if vector.is_empty() {
+                        continue;
+                    }
```

Comment:
Should we still count these with `index_point_count` _as if they were indexed_? That may make the counter less confusing. Or would that break other stuff?

Reply:
The counter won't be visible on the API until the collection is fully recovered by the WAL.
During the replay, the index upsert operations will correct this .